### PR TITLE
made the endpoint for filtering events, also new branch for easier merge

### PR DESCRIPTION
**Pull request by talosdennis**

### CODE REVIEW

File: api/Service/EventService.cs [hunk 1]
```diff
@@ -137,5 +137,27 @@ public async Task<IList<Event>> GetFutureEventList()
...
+			var eventsList = from Events in _context.Events select Events;
+
+
+			if (filtersDto.Category != null)
+			{
+				var categoriesMatching = from category in _context.Categories where (category.Title == filtersDto.Category) select category.Id;
+				eventsList = eventsList.Where(b => categoriesMatching.Contains(b.CategoryId));
+			}
+
+			if (filtersDto.EndingDate != null)
```
Comment (meda-racasan) [inline api/Service/EventService.cs]: When using nullable values, you should check if `filtersDto.EndingDate.HasValue`
Dates cannot be set to null. Read more here: http://net-informations.com/q/faq/nullable.html

File: api/Controllers/EventController.cs [hunk 2]
```diff
@@ -54,6 +54,17 @@ public async Task<IActionResult> GetFutureEvents()
 			return Ok(eventDtoList);
 		}
 
+
```
Comment (meda-racasan) [inline api/Controllers/EventController.cs]: Leave just one empty line between functions. Add one line after this function.

File: api/Service/EventService.cs [hunk 3]
```diff
@@ -137,5 +137,27 @@ public async Task<IList<Event>> GetFutureEventList()
 				 .Where(x => (x.StartingDate >= DateTime.Now)).ToListAsync();
 			return events;
 		}
+		public async Task<IList<Event>> GetFilteredEvents(FiltersDto filtersDto)
+		{
+			var eventsList = from Events in _context.Events select Events;
+
+
```
Comment (meda-racasan) [inline api/Service/EventService.cs]: Remove one extra line.

File: api/DTOs/FiltersDto.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+using System;
+namespace YonderfulApi.DTOs
+{
+  public class FiltersDto
+  {
+    [Required]
+    public DateTime StartingDate { get; set; }
+    
+    public DateTime? EndingDate { get; set; }
+    public String? Category { get; set; }
```
Comment (Mihai-Sarmaru-Yonder) [inline api/DTOs/FiltersDto.cs]: Should change this to lower case string - see here why: https://stackoverflow.com/a/7077

**Pull request merged**